vispy/visuals/surface_plot.py

```diff
@@ -116,8 +128,13 @@ def set_data(self, x=None, y=None, z=None, colors=None):
                     y = np.arange(self._z.shape[1])
                 else:
                     y = self._y
-            self.__vertices[:, :, 1] = y.reshape(1, len(y))
-            update_mesh = True
+
+            if y.ndim == 1:
+                y = y.reshape(1, len(y))
+
+            # Copy the 2D data into the appropriate slice
+            self.__vertices[:, :, 1] = y
+        update_mesh = True
```

Comment:
This `update_mesh` is under-indented.

```suggestion
            update_mesh = True
```